File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/upsert/AbstractUpsertLocation.java
package com.coupang.springframework.data.requery.domain.upsert;

import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;
import lombok.Getter;

import static com.coupang.kotlinx.core.HashxKt.hashOf;

/**
 * @author Diego on 2018. 6. 10..
 */
@Getter
@Entity(name = "UpsertLocation")
@Table(name = "upsert_location")
public abstract class AbstractUpsertLocation extends AbstractPersistable<Integer> {

    @Key
    @Generated
    @Column(name = "locationId")
    protected Integer id;

    @Column
    protected String name;

    @Embedded
    public AbstractUpsertAddress address;

    @Override
    public int hashCode() {
        return hashOf(address);
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/upsert/AbstractUpsertLocation.kt
package com.coupang.springframework.data.requery.domain.upsert

import com.coupang.kotlinx.core.hashOf
import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.*

/**
 * AbstractUpsertLocation
 *
 * @author debop@coupang.com
 * @since 18. 5. 31
 */
@Entity
@Table(name = "UpsertLocation")
abstract class AbstractUpsertLocation: AbstractPersistable<Int>() {

    @get:Key
    @get:Generated
    @get:Column(name = "locationId")
    abstract override val id: Int?

    @get:Embedded
    abstract val address: AbstractUpsertAddress

    override fun hashCode(): Int {
        return hashOf(address)
    }

}